### PR TITLE
Update report tables - Features/10475

### PR DESCRIPTION
**Pull request by FuriousLlama**
# Fixes or Implements [AB#10475](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/10475)

-   [x] Enhancement
-   [ ] Bug
-   [ ] Documentation

## Description

@FuriousLlama

- Replaced custom table with bootstrap implementation. 
- Improved loading state and cleaned up code
- Fixed issues with promises around creating a report pdf

If you are reviewing this PR, please adhere to the guidelines as [documented](https://github.com/bcgov/healthgateway/wiki/prguidance).

## Testing

-   [x] Unit Tests Updated
-   [ ] Functional Tests Updated
-   [ ] Not Required

### UI Changes

YES 

image

### CODE REVIEW

**Comment by tiagograf-AOT** [comment]
Unit tests are broken!

**Review by tiagograf-AOT — APPROVED** [review]

File: Apps/WebClient/src/ClientApp/src/components/report/medicationRequest.vue [hunk 1]
```diff
@@ -90,17 +115,55 @@ export default class MedicationRequestReportComponent extends Vue {
         );
     }
 
-    public async generatePdf(): Promise<void> {
+    public generatePdf(): Promise<void> {
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/src/components/report/medicationRequest.vue:118]: Was `async` supposed removed from the `generatePdf()` function in this file but not the ones in the other files?

File: Apps/WebClient/src/ClientApp/src/views/reports.vue [hunk 2]
```diff
@@ -207,254 +207,248 @@ export default class ReportsView extends Vue {
...
+                                class="filter-selected"
+                                title="From"
+                                data-testid="clearFilter"
+                                :pill="true"
+                                @remove="clearFilterDates"
+                            >
+                                <span data-testid="selectedDatesFilter"
+                                    >{{
+                                        reportFilter.startDate
+                                            ? `From ${formatDateLong(
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/src/views/reports.vue:266]: The `formatDateLong()` function could be removed from this file and its invocations replaced with direct calls to `DateWrapper.format()`, like you did in other files.

File: Apps/WebClient/src/ClientApp/src/views/reports.vue [hunk 3]
```diff
@@ -207,254 +207,248 @@ export default class ReportsView extends Vue {
 
 <template>
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/src/views/reports.vue:208]: I think we should have a quick chat to explain the changes to this view and to the loading component.

**Review by MikeLyttle — CHANGES_REQUESTED** [review]

File: Apps/WebClient/src/ClientApp/src/components/report/covid19.vue [hunk 4]
```diff
@@ -73,6 +83,20 @@ export default class COVID19ReportComponent extends Vue {
         return this.visibleRecords.length == 0;
     }
 
+    private get items(): LaboratoryRow[] {
+        return this.visibleRecords.map<LaboratoryRow>((x) => {
+            const labResult = x.labResults[0];
+            return {
+                date: DateWrapper.format(labResult.collectedDateTime),
```
[the remark this answers is not in the document]
Comment (FuriousLlama) [reply Apps/WebClient/src/ClientApp/src/components/report/covid19.vue:90]: Pretty interesting, didn't know that character existed. Since DateWrapper.format is used pretty much everywhere I created a story to do that and make sure it does not break anything else
https://qslvic.visualstudio.com/Health%20Gateway/_sprints/taskboard/Health%20Gateway%20Team/Health%20Gateway/Release%201.4/Sprint%202?workitem=10601

File: Apps/WebClient/src/ClientApp/test/timeline.test.ts [hunk 5]
```diff
@@ -6,6 +6,7 @@ import VueContentPlaceholders from "vue-content-placeholders";
 import VueRouter from "vue-router";
 import Vuex, { Store } from "vuex";
 
+import LoadingComponent from "@/components/loading.vue";
```
[the remark this answers is not in the document]
Comment (FuriousLlama) [reply Apps/WebClient/src/ClientApp/test/timeline.test.ts:9]: good catch. Is gone now.

File: Apps/WebClient/src/ClientApp/src/components/report/medicationRequest.vue (hunk 1 above)
Comment (FuriousLlama) [reply Apps/WebClient/src/ClientApp/src/components/report/medicationRequest.vue:118]: No, it wasn't, although AFAIK it does not make much difference since both return a promise. I have added it back

File: Apps/WebClient/src/ClientApp/src/views/reports.vue (hunk 2 above)
Comment (FuriousLlama) [reply Apps/WebClient/src/ClientApp/src/views/reports.vue:266]: good call, It's gone.

Comment (FuriousLlama) [reply Apps/WebClient/src/ClientApp/src/views/reports.vue]: Double-check - this is actually not possible since the vue template does not have access to the included libraries in the file, instead, it only has access to the methods defined in the component.
TL;DR Reverted, vue cant do that

**Review by MikeLyttle — APPROVED** [review]

**Review by tiagograf-AOT — APPROVED** [review]
